src/common/components/table/table-examples/MempoolTable.tsx

```diff
@@ -0,0 +1,391 @@
+            <ArrowsClockwise />
+          </Icon>
+        </Flex>
+      </ChakraTable.Cell>
+    </ChakraTable.Row>
+  );
+};
+
+function getToAddress(tx: MempoolTransaction): string {
+  switch (tx.tx_type) {
+    case 'contract_call':
+      return tx.contract_call?.contract_id || '';
+    case 'token_transfer':
+      return tx.token_transfer?.recipient_address || '';
+    case 'smart_contract':
+      return tx.smart_contract?.contract_id || '';
+    default:
+      return '';
+  }
+}
+
+function getAmount(tx: MempoolTransaction): number {
+  switch (tx.tx_type) {
+    case 'token_transfer':
+      return parseInt(tx.token_transfer?.amount || '0');
+    case 'contract_call':
+      return 0;
+    default:
+      return 0;
+  }
+}
+
+export function MempoolTable({
+  columnDefinitions,
+  disablePagination = false,
+  sort = 'age',
+  order = 'desc',
+  filters,
+  initialData,
+}: {
+  columnDefinitions?: ColumnDef<MempoolTableData>[];
+  disablePagination?: boolean;
+  sort?: 'age' | 'size' | 'fee';
+  order?: 'asc' | 'desc';
+  filters?: Partial<TxTableFilters>;
+  initialData?: GenericResponseType<CompressedMempoolTxTableData>;
+}) {
+  const queryClient = useQueryClient();
+  const isCacheSetWithInitialData = useRef(false);
+
+  const [pagination, setPagination] = useState<PaginationState>({
+    pageIndex: 0,
+    pageSize: TX_TABLE_PAGE_SIZE,
+  });
+
+  const { fromAddress, toAddress, transactionType } = filters || {};
+
+  if (isCacheSetWithInitialData.current === false && initialData) {
+    const queryKey = ['mempoolTransactionsInfinite', sort, order];
+    queryClient.setQueryData(queryKey, {
+      pageParams: [0],
+      pages: [initialData],
+    });
+    isCacheSetWithInitialData.current = true;
+  }
+
+  const response = useMempoolTransactionsInfinite(sort, order);
+  const txs = useInfiniteQueryResult<MempoolTransaction>(
+    response,
+    disablePagination ? undefined : TX_TABLE_PAGE_SIZE
+  );
+
+  const handlePageChange = useCallback((page: PaginationState) => {
+    setPagination(prev => ({
+      ...prev,
+      pageIndex: page.pageIndex,
+    }));
+  }, []);
+
+  const isTableFiltered =
+    filters &&
+    Object.values(filters).some(
+      filterValue => filterValue != null && filterValue !== '' && filterValue?.length !== 0
+    );
+
+  const [isSubscriptionActive, setIsSubscriptionActive] = useState(false);
+  const [newTxsAvailable, setNewTxsAvailable] = useState(false);
+
+  useSubscribeTxs(isSubscriptionActive, tx => {
+    // Waiting 5 seconds to let the API catch up to the websocket
+    setTimeout(() => {
+      setNewTxsAvailable(true);
+    }, 5000);
+    setIsSubscriptionActive(false);
+  });
+
+  useEffect(() => {
+    if (!newTxsAvailable) {
+      setIsSubscriptionActive(true);
+    }
+  }, [newTxsAvailable]);
+
+  const rowData: MempoolTableData[] = useMemo(() => {
+    let filteredTxs = txs;
+
+    if (filters) {
```

Comment:
Why are we client side filtering vs filtering in the query?

Reply:
I don't believe mempool tx endpoint supports filtering

Reply:
Do we have a task on them to add type filtering? And what about date filtering? Not sure if it's worth adding though since most txs in the mempool will be from the current day. In the meantime, I think it would make sense to remove the type filter so users aren't confused why it's only filtering on the client